src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -409,13 +412,30 @@ void OTARequestor::OnConnectionFailure(void * context, PeerId peerId, CHIP_ERROR
     default:
         break;
     }
+
+    // Give driver a chance to schedule another query
+    requestorCore->mOtaRequestorDriver->UpdateNotFound(UpdateNotFoundReason::ConnectionFailed, chip::System::Clock::Seconds32(0));
 }
 
+// Sends the QueryImage command to the next available Provider
 OTARequestorInterface::OTATriggerResult OTARequestor::TriggerImmediateQuery()
```

Comment:
When will `TriggerImmediateQuery` be called? Why would we ever call it without haven't previously set the provider location already?

Reply:
That's an application API. It's an open question whether it should exist and whether we need an API in addition/instead of this one that assumes that the provider location is set.